support-models/src/main/scala/com/gu/support/zuora/api/SubscriptionData.scala

```diff
@@ -176,7 +178,7 @@ case class Subscription(
   termType: String = "TERMED",
   readerType: ReaderType = ReaderType.Direct,
   promoCode: Option[PromoCode] = None,
-  redemptionCode: Option[RawRedemptionCode] = None,
+  redemptionCode: Option[Either[GeneratedGiftCode, RedemptionCode]] = None,
```

Comment:
this still serialises the same.  This was so the "builder" is responsible for the codes and the thing that creates the next state can get hold of the value.
TODO I think on reflection this should be reverted as there is probably a neater way